dev/com.ibm.ws.org.apache.cxf.cxf.rt.frontend.jaxws.3.2/src/org/apache/cxf/jaxws/ServiceImpl.java

```diff
@@ -177,39 +182,51 @@ private void initializePorts() {
-                        address = ((SoapAddress)e).getLocationURI();
-                    } else if (e instanceof AddressType) {
-                        address = ((AddressType)e).getLocation();
-                    } else if (e instanceof SOAP12Address) {
-                        address = ((SOAP12Address)e).getLocationURI();
-                    } else if (e instanceof SOAPAddress) {
-                        address = ((SOAPAddress)e).getLocationURI();
-                    } else if (e instanceof HTTPAddress) {
-                        address = ((HTTPAddress)e).getLocationURI();
+                for (Port port : wsdlports.values()) {
+                    QName name = new QName(serviceName.getNamespaceURI(), port.getName());
+    
+                    String address = null;
+                    String bindingID = null;
+                    List<? extends ExtensibilityElement> extensions
+                        = CastUtils.cast(port.getBinding().getExtensibilityElements());
+                    
+                    if (!extensions.isEmpty()) {
+                        ExtensibilityElement e = extensions.get(0);
+                        if (e instanceof SoapBinding) {
+                            bindingID = SOAPBinding.SOAP11HTTP_BINDING;
+                        } else if (e instanceof SOAP12Binding) {
+                            bindingID = SOAPBinding.SOAP12HTTP_BINDING;
+                        } else if (e instanceof javax.wsdl.extensions.soap.SOAPBinding) {
+                            bindingID = SOAPBinding.SOAP11HTTP_BINDING;
+                        }
+                    }
+                    if(isFinestEnabled)  {  // Liberty Change begin #26529
+                        LOG.finest("Binding id: " + bindingID);
+                    }  // Liberty Change end #26529
+                    extensions = CastUtils.cast(port.getExtensibilityElements());
+                    if (!extensions.isEmpty()) {
+                        ExtensibilityElement e = extensions.get(0);
+                        if (e instanceof SoapAddress) {
+                            address = ((SoapAddress)e).getLocationURI();
+                        } else if (e instanceof AddressType) {
+                            address = ((AddressType)e).getLocation();
+                        } else if (e instanceof SOAP12Address) {
+                            address = ((SOAP12Address)e).getLocationURI();
+                        } else if (e instanceof SOAPAddress) {
+                            address = ((SOAPAddress)e).getLocationURI();
+                        } else if (e instanceof HTTPAddress) {
+                            address = ((HTTPAddress)e).getLocationURI();
+                        }
                     }
+                    if(isFinestEnabled)  {  // Liberty Change begin #26529
+                        LOG.finest("Address: " + address);
+                    }  // Liberty Change end #26529
+                    addPort(name, bindingID, address);
```

Comment:
This addPort seems like it got moved to within the for loop, if that's the case, please move it out.

Reply:
I think it's location is correct. Because all of it's parameters such as  name, bindingID and address, they are all defined in for block.